controllers/v1/surveys.js

```diff
@@ -0,0 +1,401 @@
+   * Survey submission report
+   * @apiVersion 1.0.0
+   * @apiGroup Surveys
+   * @apiHeader {String} x-auth-token Authenticity token  
+   * @apiSampleRequest /dhiti/api/v1/surveys/submissionReport?submissionId=5f58b0b8894a0928fc8aa9b3
+   * @apiSuccessExample {json} Success-Response:
+   * {
+   *   "solutionName": "",
+       "response": [{
+         "order": "",
+         "question": "",
+         "responseType": "",
+         "answers": [],
+         "chart": {},
+         "instanceQuestions":[],
+         "evidences":[
+              {"url":"", "extension":""}
+          ]
+       }]
+   * }
+   * @apiUse errorBody
+   */
+
+   exports.submissionReport = async function (req, res) {
+
+    return new Promise(async function (resolve, reject) {
+
+        if (!req.query.submissionId) {
+
+            let response = {
+                result: false,
+                message: 'submissionId is a required field'
+            };
+            res.send(response);
+
+        } else {
+
+            model.MyModel.findOneAsync({ qid: "survey_submission_report_query" }, { allow_filtering: true })
+                .then(async function (result) {
+
+                    let bodyParam = JSON.parse(result.query);
+
+                    if (config.druid.survey_datasource_name) {
+                        bodyParam.dataSource = config.druid.survey_datasource_name;
+                    }
+
+                    bodyParam.filter.fields[0].value = req.query.submissionId;
+
+                     //if filter is given
+                     if (req.body.filter && req.body.filter.questionId && req.body.filter.questionId.length > 0) {
+                        let filter = { "type": "in", "dimension": "questionExternalId", "values": req.body.filter.questionId };
+                        bodyParam.filter.fields.push(filter);
+                    }
+                    else {
+                        let filter = { "type": "not", "field": { "type": "selector", "dimension": "questionAnswer", "value": "" } };
+                        bodyParam.filter.fields.push(filter);
+                    }
+
+                    //pass the query as body param and get the resul from druid
+                    let options = config.druid.options;
+                    options.method = "POST";
+                    options.body = bodyParam;
+                    let data = await rp(options);
+
+                    if (!data.length) {
+
+                        let message;
+                        let getSubmissionStatusResponse = await assessmentService.getSurveySubmissionStatusById
+                        (
+                            req.query.submissionId
+                        )
+
+                        if (getSubmissionStatusResponse.result && 
+                            getSubmissionStatusResponse.result.status == filesHelper.submission_status_completed) {
+                            message = "Report is being created, please check after a while"
```

Comment:
@deepakr123 - Change the text to - "Your report generation is in progress. Please check back after a few minutes."

Reply:
resolved